src/Surfnet/StepupRa/RaBundle/Security/Authentication/Token/SamlToken.php

```diff
@@ -61,6 +63,34 @@ public function getInstitutionConfigurationOptions()
     }
 
     /**
+     * @param string $institution
+     * @param InstitutionConfigurationOptions $institutionConfigurationOptions
+     */
+    public function changeInstitutionScope(
+        $institution,
+        InstitutionConfigurationOptions $institutionConfigurationOptions
+    ) {
+        if ($this->getUser() === null) {
+            throw new LogicException('Cannot change institution scope: token does not contain a user');
+        }
+
+        $roles = array_map(function (RoleInterface $role) {
+            return $role->getRole();
+        }, $this->getRoles());
+
+        if (!in_array('ROLE_SRAA', $roles)) {
+            throw new ForbiddenException(sprintf(
```

Comment:
I have grown accustomed to adding specific runtime exceptions as they tend to clarify what specifically has gone wrong. The message is a further explanation of what has gone wrong. Usually custom exceptions have benefits for logging and fine-grained exception handling as well. As this is not the case here, we might as well remove it.